packages/kolibri-zip/src/fileUtils.js

```diff
@@ -0,0 +1,46 @@
+export function getAbsoluteFilePath(baseFilePath, relativeFilePath) {
+  // Construct a URL with a dummy base so that we can concatenate the
+  // dependency URL with the URL relative to the dependency
+  // and then read the pathname to get the new path.
```

Comment:
I realize this was pre-existing and it's just moving it, so non-blocking.

This comment feels rather hard to wrap my brain around, mostly the 
`that we can concatenate the dependency URL with the URL relative to the dependency` part 
The second "URL" (i.e. "the URL relative to the dependency") is where I get lost.

Reply:
Basically, we are using the JS URL utilities to resolve a relative file path (by turning it into a resolution of a relative URL against a dummy base URL).

So the file that we are currently looking at in the zip file: e.g. `style/images.css` references another file relatively `../images/cool.png`, this will resolve the reference to an absolute file path from the root of the zip file - so `images/cool.png`.

Reply:
This is a bit confusing for me as well but I played w/ it a little bit and I think that the thing that isn't clear is that the `URL` constructor auto-magically resolves the relative path against the second argument.

I played w/ it in the terminal a bit and this kind of made it make sense to me as I kept adding `..`s to the relative path:

```
> new URL("../best/friend.css", "https://a.com/hi/there/my/mortal/enemy").pathname
'/hi/there/my/best/friend.css'
> new URL("../../best/friend.css", "https://a.com/hi/there/my/mortal/enemy").pathname
'/hi/there/best/friend.css'
> new URL("../../../best/friend.css", "https://a.com/hi/there/my/mortal/enemy").pathname
'/hi/best/friend.css'
```

I think that maybe the comment could use some clarification or an example like the one you gave, Richard, re: `styles/header/logo.css` has a reference to `../../assets/logo.svg` so we need to get a pathname `assets/logo.svg` out of that. Maybe that's more complicated though 😄